aws/hll/aws-s3-transfer-manager/examples/cp.rs

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+use std::error::Error;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::{mem, time};
+
+use aws_s3_transfer_manager::download::Downloader;
+
+use aws_s3_transfer_manager::download::body::Body;
+use aws_sdk_s3::operation::get_object::builders::GetObjectInputBuilder;
+use clap::{CommandFactory, Parser};
+use tokio::fs;
+use tokio::io::AsyncWriteExt;
+
+type BoxError = Box<dyn Error + Send + Sync>;
+
+const ONE_GIGABYTE: u64 = 1024 * 1024 * 1024;
+
+#[derive(Debug, Clone, clap::Parser)]
+#[command(name = "cp")]
+#[command(about = "Copies a local file or S3 object to another location locally or in S3.")]
+pub struct Args {
+    /// Source to copy from <S3Uri | Local>
+    #[arg(required = true)]
+    source: TransferUri,
+
+    /// Destination to copy to <S3Uri | Local>
+    #[arg(required = true)]
+    dest: TransferUri,
+
+    /// Number of concurrent uploads/downloads to perform.
+    #[arg(long, default_value_t = 8)]
```

Comment:
should we make the default [`available_parallelism`](https://doc.rust-lang.org/std/thread/fn.available_parallelism.html)? there is some [precedent of this being done](https://github.com/libcpr/cpr/blob/master/include/cpr/threadpool.h#L16).

Reply:
In async Rust, you can have 1 thread handling **MANY** concurrent HTTP requests.